### PR TITLE
customGear and fauxWeapons rebasing

### PR DESCRIPTION
**Pull request by AChesheireCat**
Adds a new addon "customGear" which is meant to act as a gathering point for all "new" gear/weapon items that we may want to add to potato. This'll also help clean up miscFixes since I've kinda cluttered it with things that aren't really fixes, but also aren't fit anywhere else.

As such, I've rebased the faux weapons that sat in miscFixes to this new addon. 

I've also added a new thing that adds a copy of certain CUP vests as backpacks for mission makers and loadout makers to play around with.

customGear should now include the following:

- fauxWeapons
    - fauxC7
    - fauxMG5
- vestBackpacks
    - BAF Mk4 belt and webbing
    - RUAF Smersh in olive and coyote

### CODE REVIEW

File: addons/customGear/fauxWeapons/config.cpp [hunk 1]
```diff
@@ -81,4 +106,21 @@ class CfgWeapons {
             maxRangeProbab = 0.05;
         };
     };
-};
\ No newline at end of file
+    
+    // Adds MG5 in 7.62x51mm
+    class MMG_01_tan_F;
+    class GVAR(MG5): MMG_01_tan_F {
```
Comment (lambdatiger) [inline addons/customGear/fauxWeapons/config.cpp:112]: You shouldn't generally change a class name that some one might be using once it actually enters the public version of a mod, this includes the C7 (currently `potato_fauxC7_c7`) and MG5 (currently `potato_fauxMG5_MG5`). That's also why I was asking if you'd be willing to choose a different prefix here https://github.com/BourbonWarfare/POTATO/pull/590#pullrequestreview-2393983513.

**Review by lambdatiger — COMMENTED** [review]
Looks ok from what I can see, some feedback on structure and naming that will come up later.

File: addons/customGear/fauxWeapons/config.cpp
[the remark this answers is not in the document]
Comment (AChesheireCat) [reply addons/customGear/fauxWeapons/config.cpp:1]: donezo

File: addons/customGear/fauxWeapons/config.cpp (hunk 1 above)
Comment (AChesheireCat) [reply addons/customGear/fauxWeapons/config.cpp]: i think i fixed this now, but i agree - i'll update the kac lamg branch to have a set classname following the "potato_(weaponclass)_(weaponname)" format

File: addons/customGear/config.hpp [hunk 2]
```diff
@@ -2,18 +2,16 @@
 
 class CfgPatches {
     class ADDON {
-        weapons[] = {
-            QGVAR(MG5)
-        };
         units[] = {};
+        weapons[] = {};
```
Comment (PabstMirror) [inline addons/customGear/config.hpp:6]: we might want to put weapons in here, 
i don't think it matters for arsenal but I think it does for the inventory box editor in eden/zeus?

Comment (AChesheireCat) [reply addons/customGear/config.hpp]: done!

**Comment by AChesheireCat** [comment]
Should the "baseWeapon = QGVAR(MG5)" line be changed to baseWeapon = "potato_fauxMG5_MG5"? Or is it fine as it is?

**Comment by lambdatiger** [comment]
> Should the "baseWeapon = QGVAR(MG5)" line be changed to baseWeapon = "potato_fauxMG5_MG5"? Or is it fine as it is?

Yes

**Review by PabstMirror — APPROVED** [review]

**Pull request merged**